test/MarginAccount.test.ts

```diff
@@ -0,0 +1,496 @@
+        assert.equal(vault.shortAmounts[1], new BigNumber(11))
+      })
+
+      it('should add some more of the second short otoken', async () => {
+        await marginAccountTester.testAddShort(otoken2.address, 12, 1)
+        const vault = await marginAccountTester.getVault()
+        assert.equal(vault.shortAmounts[1], new BigNumber(23))
+      })
+
+      it('should revert if trying to add wrong otoken to an index', async () => {
+        await expectRevert(
+          marginAccountTester.testAddShort(otoken.address, 10, 1),
+          'MarginAccount: invalid short otoken position',
+        )
+      })
+
+      // it('should revert if trying on the wrong vault', async () => {
+      //   // await expectRevert(marginAccountTester.testAddShort(otoken.address, 10, 5, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 3, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 100, {from: deployer}), 'revert')
+      // })
+    })
+
+    describe('Add long', () => {
+      it('vaultIds should be zero, owner should be null', async () => {
+        account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.owner, testAccount.owner, 'MarginAccount.Account owner addr mismatch')
+      })
+
+      it('should add long otokens', async () => {
+        const index = 0
+        const amount = 10
+        await marginAccountTester.testAddLong(otoken.address, amount, index)
+        const vault = await marginAccountTester.getVault()
+        assert.equal(vault.longAmounts[index], new BigNumber(amount))
+      })
+
+      it('should add a different long otoken', async () => {
+        const index = 1
+        const amount = 10
+        await marginAccountTester.testAddLong(otoken2.address, amount, index)
+        const vault = await marginAccountTester.getVault()
+        assert.equal(vault.longAmounts[index], new BigNumber(amount))
+      })
+
+      it('should revert if trying to add wrong otoken to an index', async () => {
+        const index = 1
+        const amount = 10
+        await expectRevert(
+          marginAccountTester.testAddLong(otoken.address, amount, index),
+          'MarginAccount: invalid long otoken position',
+        )
+      })
+
+      //TODO: do we need to check addition overflows?
+      // //TODO: should it revert if trying to add a long otoken to an vault where the same otoken is shorted?'
+      // it('should revert if trying on the wrong vault', async () => {
+      //   // await expectRevert(marginAccountTester.testAddShort(otoken.address, 10, 5, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 3, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 100, {from: deployer}), 'revert')
+      // })
```

Comment:
yes this is a good thing to test!

Reply:
we are using SafeMath right, why we need to check addition overflows ?

Reply:
don't need this IMO